File: src/main/java/org/home/todobackend/service/StatService.java
package org.home.todobackend.service;

import org.home.todobackend.entity.Stat;
import org.home.todobackend.repo.StatRepository;
import org.springframework.stereotype.Service;

import javax.transaction.Transactional;

@Service
@Transactional
public class StatService {
    private final StatRepository repository;
    public StatService(StatRepository repository) {
        this.repository = repository;
    }
    public Stat findStat(String email) {
        return repository.findByUserEmail(email);
    }
}

File: src/main/java/org/home/todobackend/service/StatService.kt
package org.home.todobackend.service

import org.home.todobackend.repo.StatRepository
import org.home.todobackend.entity.Stat
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional
class StatService(private val repository: StatRepository) {
    fun findStat(email: String): Stat {
        return repository.findByUserEmail(email)
    }
}